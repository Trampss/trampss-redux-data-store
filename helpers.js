!function(t,e){"object"==typeof exports&&"undefined"!=typeof module?e(exports):"function"==typeof define&&define.amd?define(["exports"],e):e(t["trampss-redux-factory"]=t["trampss-redux-factory"]||{})}(this,function(t){"use strict";var e=Object.assign||function(t){for(var e=1;e<arguments.length;e++){var n=arguments[e];for(var r in n)Object.prototype.hasOwnProperty.call(n,r)&&(t[r]=n[r])}return t};t.payloadWrapper=function(t){return function(n){return function(){return function(){return function(r){var o=r.action,a=o.payload,u=o.type;return!t||t.test(u)?e({},r,{action:e({},r.action,{payload:n?n(a):a})}):r}}}}},Object.defineProperty(t,"__esModule",{value:!0})});
//# sourceMappingURL=helpers.js.map
