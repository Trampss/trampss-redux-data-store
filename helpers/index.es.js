function n(n,t,r){return t in n?Object.defineProperty(n,t,{value:r,enumerable:!0,configurable:!0,writable:!0}):n[t]=r,n}function t(t){for(var r=1;r<arguments.length;r++){var e=null!=arguments[r]?arguments[r]:{},u=Object.keys(e);"function"==typeof Object.getOwnPropertySymbols&&(u=u.concat(Object.getOwnPropertySymbols(e).filter(function(n){return Object.getOwnPropertyDescriptor(e,n).enumerable}))),u.forEach(function(r){n(t,r,e[r])})}return t}var r=function(n){return function(){return function(){return function(){return function(){return function(r){return t({},r,{state:n(r.action,r.state)})}}}}}},e=function(n){return function(){return function(){return function(){return function(){return function(r){return t({},r,{action:n(r.action)})}}}}}},u=function(n){return function(t){return function(){return function(){return function(){return function(){return function(r){var e=r.action;return!n||n.test(e.type)?{action:e,state:t(r.state)}:r}}}}}}},o=function(n){return function(r){return function(){return function(){return function(){return function(){return function(u){var o=u.action,c=o.payload;return!n||n.test(o.type)?e(function(n){return t({},n,{payload:r(c)})})()()()()(u):u}}}}}}};export{r as reducer,e as mapAction,u as mapState,o as mapPayload};
