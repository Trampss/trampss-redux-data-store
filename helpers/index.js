!function(t,n){"object"==typeof exports&&"undefined"!=typeof module?n(exports):"function"==typeof define&&define.amd?define(["exports"],n):n(t["trampss-redux-factory"]=t["trampss-redux-factory"]||{})}(this,function(t){"use strict";var n=Object.assign||function(t){for(var n=1;n<arguments.length;n++){var e=arguments[n];for(var r in e)Object.prototype.hasOwnProperty.call(e,r)&&(t[r]=e[r])}return t};t.mapAction=function(t){return function(){return function(){return function(e){return n({},e,{action:t(e.action)})}}}},t.mapPayload=function(t){return function(e){return function(){return function(){return function(r){var o=r.action,u=o.payload,i=o.type;return!t||t.test(i)?n({},r,{action:n({},r.action,{payload:e(u)})}):r}}}}},Object.defineProperty(t,"__esModule",{value:!0})});
//# sourceMappingURL=index.js.map
