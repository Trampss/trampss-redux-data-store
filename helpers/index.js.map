{"version":3,"file":"index.js","sources":["../src/helpers.js"],"sourcesContent":["export const reducer = mapper =>\n  // middleware signature\n  (/* key */) => (/* prefix */) => (/* name */) => (/* defaultData */) => ctx => ({ ...ctx, state: mapper(ctx.action, ctx.state) })\n\n\nexport const mapAction = mapper =>\n  // middleware signature\n  (/* key */) => (/* prefix */) => (/* name */) => (/* defaultData */) => ctx => ({\n    ...ctx,\n    action: mapper(ctx.action),\n  })\n\nexport const mapState = actionMatches => mapper =>\n  // middleware signature\n  (/* key */) => (/* prefix */) => (/* name */) => (/* defaultData */) => (ctx) => {\n    const action = ctx.action\n    if (!actionMatches || actionMatches.test(action.type)) {\n      return { action, state: mapper(ctx.state) }\n    }\n    return ctx\n  }\n\nexport const mapPayload = actionMatches => mapper =>\n  // middleware signature\n  (/* key */) => (/* prefix */) => (/* name */) => (/* defaultData */) => (ctx) => {\n    const { payload, type } = ctx.action\n    if (!actionMatches || actionMatches.test(type)) {\n      return mapAction(action => ({ ...action, payload: mapper(payload) }))()()()(ctx)\n    }\n    return ctx\n  }\n"],"names":["mapAction","ctx","mapper","action","state","actionMatches","test","type","payload"],"mappings":"6WAKaA,EAAY,mHAGlBC,UACKC,EAAOD,EAAIE,0BATA,mHAEgEF,GAAKG,MAAOF,EAAOD,EAAIE,OAAQF,EAAIG,wCAUlG,mBAAiB,oGAEkCH,OACjEE,EAASF,EAAIE,cACdE,GAAiBA,EAAcC,KAAKH,EAAOI,OACrCJ,SAAQC,MAAOF,EAAOD,EAAIG,QAE9BH,sBAGe,mBAAiB,oGAEgCA,SAC7CA,EAAIE,OAAtBK,IAAAA,QAASD,IAAAA,YACZF,GAAiBA,EAAcC,KAAKC,GAChCP,EAAU,wBAAgBG,GAAQK,QAASN,EAAOM,aAAmBP,GAEvEA"}