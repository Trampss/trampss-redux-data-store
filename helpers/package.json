{
  "name": "trampss-redux-factory/helpers",
  "private": true,
  "main": "../helpers.js"
}
