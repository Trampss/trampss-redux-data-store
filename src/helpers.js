export const reducer = mapper =>
  // middleware signature
  (/* key */) => (/* prefix */) => (/* name */) => (/* defaultData */) => ctx => ({ ...ctx, state: mapper(ctx.action, ctx.state) })


export const mapAction = mapper =>
  // middleware signature
  (/* key */) => (/* prefix */) => (/* name */) => (/* defaultData */) => ctx => ({
    ...ctx,
    action: mapper(ctx.action),
  })

export const mapState = actionMatches => mapper =>
  // middleware signature
  (/* key */) => (/* prefix */) => (/* name */) => (/* defaultData */) => (ctx) => {
    const action = ctx.action
    if (!actionMatches || actionMatches.test(action.type)) {
      return { action, state: mapper(ctx.state) }
    }
    return ctx
  }

export const mapPayload = actionMatches => mapper =>
  // middleware signature
  (/* key */) => (/* prefix */) => (/* name */) => (/* defaultData */) => (ctx) => {
    const { payload, type } = ctx.action
    if (!actionMatches || actionMatches.test(type)) {
      return mapAction(action => ({ ...action, payload: mapper(payload) }))()()()(ctx)
    }
    return ctx
  }
