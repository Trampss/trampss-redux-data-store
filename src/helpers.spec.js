/* eslint-env jest */
import { mapAction, mapPayload } from './helpers'

const ctx = {
  action: {
    payload: 'payload',
    type: 'TYPE',
  },
  state: {},
}

const mapperAction = () => 'ACTION MAPPED !'
const mapperPayload = () => 'PAYLOAD MAPPED !'

describe('helpers/mapPayload', () => {
  it('should map payload without matching type', () => expect(mapPayload()(mapperPayload)()()(ctx)).toMatchSnapshot())
  it('should not map payload (type matcher is KO)', () => expect(mapPayload(/NOTHING/)(mapperPayload)()()(ctx)).toMatchSnapshot())
  it('should map payload (type matcher OK)', () => expect(mapPayload(/TYPE/)(mapperPayload)()()(ctx)).toMatchSnapshot())
})

describe('helpers/mapAction', () => {
  it('should map action', () => expect(mapAction(mapperAction)()()(ctx)).toMatchSnapshot())
})
