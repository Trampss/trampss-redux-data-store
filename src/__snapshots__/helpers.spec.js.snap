// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers/mapAction should map action 1`] = `
Object {
  "action": "ACTION MAPPED !",
  "state": Object {},
}
`;

exports[`helpers/mapPayload should map payload (type matcher OK) 1`] = `[Function]`;

exports[`helpers/mapPayload should map payload without matching type 1`] = `[Function]`;

exports[`helpers/mapPayload should not map payload (type matcher is KO) 1`] = `
Object {
  "action": Object {
    "payload": "payload",
    "type": "TYPE",
  },
  "state": Object {},
}
`;

exports[`helpers/mapState should map state (type matcher OK) 1`] = `
Object {
  "action": Object {
    "payload": "payload",
    "type": "CUSTOM",
  },
  "state": "STATE MAPPED !",
}
`;

exports[`helpers/mapState should map state (type matcher is KO) 1`] = `
Object {
  "action": Object {
    "payload": "payload",
    "type": "CUSTOM",
  },
  "state": "STATE",
}
`;

exports[`helpers/mapState should map state without matching type 1`] = `
Object {
  "action": Object {
    "payload": "payload",
    "type": "CUSTOM",
  },
  "state": "STATE MAPPED !",
}
`;

exports[`helpers/reducer should add reducer but not change state 1`] = `
Object {
  "action": Object {
    "payload": "payload",
    "type": "TYPE",
  },
  "state": Object {},
}
`;

exports[`helpers/reducer should add reducer to change state 1`] = `
Object {
  "action": Object {
    "payload": "payload",
    "type": "CUSTOM",
  },
  "state": "CUSTOM_STATE",
}
`;
